### PR TITLE
3DES CMAC Support

### PR DESCRIPTION
**Pull request by bmurray7**
Added support for 3DES CMAC.

I'm having trouble understanding why `tests/scripts/check-generated-files.sh` is failing on the continuous integration build.  

If anyone has any suggestions, I'll promptly make the appropriate fixes.

### CODE REVIEW

**Comment by bmurray7** [comment]
@mpg I've having trouble understanding why `tests/scripts/check-generated-files.sh` is failing in continuous integration testing.  It seems to work on my machine.  Do you have any suggestions?

**Comment by bmurray7** [comment]
@mpg Do you have any suggestions about the issue with `tests/scripts/check-generated-files.sh`?

**Comment by pjbakker** [comment]
Hey @bmurray7,
 
@mpg is not actively working on mbed TLS anymore.

In 7bc7d00 you modify error.c
Is that intended? (It's the reason the check fails as now the order has changed)

File: library/cmac.c [hunk 1]
```diff
@@ -273,40 +332,53 @@ int mbedtls_cmac_verify( mbedtls_cmac_context *ctx,
...
+    exit:
+        mbedtls_free( check_tag );
+        return( ret );
 }
 
+#ifdef MBEDTLS_AES_C
 /*
  * PRF based on CMAC with AES-128
- * TODO: add reference to the standard
- * TODO: do we need to take a cmac_context as an argument here?
```
Comment (bmurray7) [inline library/cmac.c]: No, `cmac_context` is not needed.

**Pull request closed without merging**